dlk/tests/test_packer.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 The Blueoil Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+"""Unittest for Packer."""
+
+import unittest
+
+from modules.packer import Packer
+
+import numpy as np
+
+
+class TestPacker(unittest.TestCase):
+    """Test class for Packer."""
+
+    def test_bw1_dividable_by_wordsize(self):
+        """Test for when the input tensor size is able to divide by wordsize (1 bit version)."""
+        packer = Packer(1, 32)
+
+        test_input = np.zeros([32], dtype=np.float32)
+        test_input[0:6] = [0, 1, 0, 1, 0, 1]
+
+        test_output = packer.run(test_input)
+        print(test_output)
+        self.assertEqual(test_output[0], 42)
+
+    def test_bw2_dividable_by_wordsize(self):
+        """Test for when the input tensor size is able to divide by wordsize (2 bit version)."""
+        packer = Packer(2, 32)
+
+        test_input = np.zeros([32], dtype=np.float32)
+        test_input[0:6] = [0, 3, 0, 3, 0, 3]
+
+        test_output = packer.run(test_input)[0]
+
+        self.assertEqual(test_output[0], 42)
+        self.assertEqual(test_output[1], 42)
+
+    def test_bw2_not_dividable_by_wordsize(self):
+        """Test for when the input tensor size is not able to divide by wordsize."""
+        packer = Packer(2, 32)
+
+        test_input = np.zeros([37], dtype=np.float32)
+        test_input[0:-1:2] = 1
+        test_input[0:-1:4] = 2
+
+        test_output = packer.run(test_input)[0]
+        expectged_output = np.array([1145324612, 286331153, 4294967268, 4294967265], dtype=np.uint32)
+
+        self.assertTrue((test_output == expectged_output).all())
```

Comment:
Maybe it's better to use `assert_array_equal`?
https://docs.scipy.org/doc/numpy/reference/generated/numpy.testing.assert_array_equal.html#numpy.testing.assert_array_equal

Reply:
I choose `.all()` here, to be consisitent to other tests.